(sourcetree) Use MSI installer

pascalberger opened the pull request:
## Description
Use new enterprise version of Sourcetree which comes as an MSI

## Motivation and Context
The current Squirrel based installer is not really suitable for a deployment with Chocolatey as it installs in the user directory. Using the MSI fixes these issues.

Fixes #1001

## How Has this Been Tested?
Tested locally

## Types of changes
<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Migrated package (a package has been migrated from another repository)

## Checklist:
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this repository.
- [x] My change requires a change to documentation (this usually means the notes in the description of a package).
- [ ] I have updated the documentation accordingly (this usually means the notes in the description of a package).
- [x] All files are up to date with the latest [Contributing Guidelines](https://github.com/chocolatey/chocolatey-coreteampackages/blob/master/CONTRIBUTING.md)
- [ ] The added/modified package passed install/uninstall in the chocolatey test environment.
- [x] The changes only affect a single package (not including meta package).

pascalberger commented:
Some questions by myself:

The MSI installs side by side with the Squirrel based installer used in current versions of this package. Since using the Squirrel based installer no longer makes sense I don't think the MSI deserves it's own package, but instead the existing package should be updated. But currently there is no proper upgrade path from previous versions to this version. Should we try to uninstall existing Sourcetree installations in this package? What if a user really want to have both installed side by side?

Settings also want be migrated from existing installations. Should we try to do this migration or is this out of scope for a Chocolatey package?

I also should update the description in the nuspec to reflect the above points based on the direction we want to take.

AdmiringWorm commented:
@pascalberger could you rebase towards the latest of master? Currently there are merge conflicts.

I'll review the PR after that have been done.

AdmiringWorm commented on automatic/sourcetree/tools/chocolateyInstall.ps1:
```diff
@@ -1,17 +1,17 @@
+  checksumType64= 'sha256'
 }
 
 Install-ChocolateyPackage @packageArgs
```
what about updates, is the new MSI installer subjected to the same kind of error during update as quite a few MSIs have?

pascalberger commented:
@AdmiringWorm @gep13 I've now added a check if there's already an existing installation of Sourcetree installed using the Squirrel installer on the machine. If so it asks the user if it should be uninstalled. Settings will **not** be migrated to the new installation.

This is ready for a review from my side

pascalberger commented:
> You can detect previous existing package and throw with adequate message for a user. User should devise migration strategy if thats needed.

@majkinetor Currently the package checks for existing Squirrel installation and asks user if he wants to uninstall it. Does this fit with what you asked? Or do you prefer to only print out a note that another SourceTree installation is present and new package will be installed side-by-side?

pascalberger commented:
@majkinetor I removed the uninstall code and just output a warning that standard version of Sourcetree won't be uninstalled. Also added an additional note to the package description

majkinetor commented:
Is there anything else you need to do here ?
It would be good to have some proof that this works before merging. Could you `Test-Package` locally or inside VM and give output here (or something equivalent).

majkinetor commented:
Nvm merging this, I thought AV buid failed for different reasons.

majkinetor commented:
Hm... I see now that you didn't throw here. Please throw on older version found. Nobody will see warnings once they do `cup all` but they will see errors as they are listed at the end of all updates. Nobody expects this - installing the same named software and getting something parallel to existing one.

We have similar situation with VSCode x64 install, and package didn't hendled it, but during installl VSCode would complain, which was singnal to all users that intervention is required.

pascalberger commented:
@majkinetor I don't think throwing will lead to the best user experience. This forces the user to manually uninstall his prior Sourcetree installation before he can install the new version. By this he loses all settings. By only warning he at least can check the settings in the old version and apply them in the new version before uninstalling the old version.

majkinetor commented:
Well... ok...
